### ISSUE TITLE
OSError when converting and/or saving binaries

### ISSUE THREAD

**Issue by marysethomas** [MouseLand/suite2p#318]
Hello,

I commonly run multiple recording blocks together (>50,000 tifs) but I have recently started experiencing an issue where I can't get past the "converting to binary" step because an OSError is thrown very late in the conversion process (possibly during saving but not always). I started using the GitHub version of suite2p, which appeared to solve the issue for single blocks, but it still happens for multiple blocks. I have been unable to determine if this is related to the number of tifs I am loading at once.

I believe the error may have to do with ScanImageTiffReader and it takes the form: OSError: exception: access violation reading 0x0000000000002A

I'm wondering if anyone else has experience with this error or words of advice?

See screenshot below:

![Screen Shot 2020-04-22 at 12 09 15 PM](https://user-images.githubusercontent.com/7726560/80005834-1fb0e600-8492-11ea-8745-94c203f2af90.png)

**Comment by marysethomas** [comment]
Hello, just an update to my issue. I started experiencing the error for smaller datasets (i.e. single blocks) again, and was able to get everything working only after I completely deleted the suite2p directory and reinstalled it. Removing the suite2p environment was not enough. I have not confirmed whether I still have the issue for large datasets yet.

Where the issue arises is still a mystery to me, but at least this has allowed me to keep working!

**Comment by carsen-stringer** [comment]
very weird, this looks like a reading issue. maybe your disk is getting slowed down? Could be that deleting everything meant that you installed a newer ScanImageTiffReader package that doesn't have this error :) going to close this issue for now, thanks for the detailed info

**Comment by marysethomas** [comment]
Hi @carsen-stringer, thanks for your reply.

Unfortunately, this didn't turn out to be a permanent fix. The problem has now come back and it happens for small ~4000 tif files again.

Do you have any suggestions for troubleshooting?

**Comment by carsen-stringer** [comment]
are you connected to a server? do you think it's intermittent file-reading dropping? scanimagetiffreader has not been updated in over a year so I don't think it's their package

**Comment by marysethomas** [comment]
It happens on my local drive as well as when I'm working off a server. It definitely seems like it could be an intermittent issue with file-reading. However, it's strange to me that deleting and reinstalling the suite2p directory appears to "reset" the issue! I'll continue looking into this on my end.

Is it possible to combine binary files after they've been created? This would be a nice feature to allow users to process their data in smaller batches.

**Comment by carsen-stringer** [comment]
no that's not something we will support in the near future, but if you write a script or jupyter-notebook we'll be happy to add it to the repo. glad it sounds like it's working now

**Comment by marysethomas** [comment]
Ok, thanks for your help again!

**Comment by marysethomas** [comment]
Problem solved! I'm commenting here in case this will help anyone else.

The issue was that I had `look_one_level_down = true` and some of my data had subfolders with images that were not supposed to be included. I'm guessing ScanImageTiffReader ran into issues with trying to combine these images with the rest of my tiffs.

Deleting my directory helped by resetting my default Ops, and I associated the error with large datasets where I needed to use `look_one_level_down = true` in order to view all the directories in the Run Suite2p GUI.

Thanks for your patience Carsen!
